presto-main-base/src/test/java/com/facebook/presto/sql/relational/TestRowExpressionOptimizer.java

```diff
@@ -292,14 +280,96 @@ public void testFunctionNotInPrestoDefaultNamespaceIsNotEvaluated()
         assertEquals(optimizedCall.getChildren().get(0), constant(8.0, DOUBLE));
     }
 
-    private static RowExpression ifExpression(RowExpression condition, long trueValue, long falseValue)
+    @Test
+    public void testSpecialFormExpressionsWhenDefaultNamespaceIsSwitched()
     {
-        return new SpecialFormExpression(IF, BIGINT, ImmutableList.of(condition, constant(trueValue, BIGINT), constant(falseValue, BIGINT)));
+        RowExpressionOptimizer nativeOptimizer = getNativeOptimizer();
+
+        RowExpression leftCondition = callOperator(
+                GREATER_THAN,
+                SQUARE_ROOT_EXP,
+                constant(5.0, DOUBLE));
+
+        RowExpression rightCondition = callOperator(
+                LESS_THAN,
+                CUBE_ROOT_EXP,
+                constant(10.0, DOUBLE));
+
+        RowExpression andExpr = new SpecialFormExpression(
+                SpecialFormExpression.Form.AND,
+                BOOLEAN,
+                ImmutableList.of(leftCondition, rightCondition));
+
+        RowExpression optimized = nativeOptimizer.optimize(andExpr, OPTIMIZED, SESSION);
+        assertEquals(optimized, constant(true, BOOLEAN));
```

Comment:
Do you want to add a negative test like the ones above

`assertThrows(IllegalArgumentException.class, () -> optimizer.optimize(andExpr, OPTIMIZED, SESSION));
`

Reply:
That needs a static expression. Skipping for now.